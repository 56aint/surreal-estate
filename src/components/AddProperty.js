import React, { useState } from "react";
import "../styles/AddProperty.css";
import Alert from "./Alert";

const axios = require("axios");

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "",
      type: "",
      bedrooms: "",
      bathrooms: "",
      price: "",
      email: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const handleAddProperty = (event) => {
    event.preventDefault();
    // console.log(fields);
    setAlert({ message: "", isSuccess: false });
    axios
      .post(`http://localhost:4000/api/v1/PropertyListing`, {
        ...fields,
      })
      .then((response) => {
        console.log(response);
        setAlert({
          message: "Property Added",
          isSuccess: true,
        });
      })
      .catch((error) => {
        console.log(error);
        setAlert({
          message: "Server error, Please try again later",
          isSuccess: false,
        });
      });
  };

  const handleFieldChange = (event) => {
    const changeField = event.target.name;
    const newValue = event.target.value;
    setFields({ ...fields, [changeField]: newValue });
    console.log(fields);
  };

  return (
    <>
      <div className="add-property">
        <form onSubmit={handleAddProperty}>
          <Alert message={alert.message} success={alert.isSuccess} />
          <label htmlFor="title" className="form-label">
            Title
            <input
              placeholder="Property Description"
              type="text"
              id="title"
              name="title"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>

          <label htmlFor="city" className="form-label">
            <select
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="">Choose a city</option>
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Shefield">Shefield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>

          <label htmlFor="type" className="form-label">
            <select
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="">Choose a Property type</option>
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
          <label htmlFor="bedrooms" className="form-label">
            Bedrooms
            <input
              className="number-input"
              type="number"
              id="bedrooms"
              name="bedrooms"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>

          <label htmlFor="bathrooms" className="form-label">
            Bathrooms
            <input
              className="number-input"
              type="number"
              id="bathrooms"
              name="bathrooms"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>

          <label htmlFor="price" className="form-label">
            Price
            <span className="gbp">
              <input
                className="number-input"
                placeholder="Enter whole GBP (&pound;)"
                type="number"
                id="price"
                name="price"
                value={fields.price}
                onChange={handleFieldChange}
              />
            </span>
          </label>
          <br />
          <label htmlFor="email" className="form-label">
            Email
            <input
              className="email-input"
              placeholder="your email"
              type="email"
              id="email"
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
          <button type="submit" className="add-property-button">
            Add
          </button>
        </form>
      </div>
    </>
  );
};

export default AddProperty;
