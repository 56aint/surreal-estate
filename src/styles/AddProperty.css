.add-property {
  padding: 10px;
  margin: 10px;
}

.add-property-button {
  background: rgb(151, 3, 3);
  color: white;
  padding: 3px 20px;
  margin: 10px;
  font-size: 12px;
  cursor: pointer;
  border: none;
  transition: all 0.5s;
  /*display: inline-block;*/
}
.add-property-button {
  cursor: pointer;
  display: inline-block;
  position: relative;
}
.add-property-button {
  padding: 2px 15px;
  font-size: 16px;
  text-align: center;
  cursor: pointer;
  outline: none;
  color: #fff;
  background: rgb(151, 3, 3);
  border: none;
  border-radius: 15px;
  box-shadow: 0 9px #999;
}

.add-property-button:hover {background: rgba(238, 13, 13, 0.75);}

.add-property-button:active {
  background: rgba(238, 13, 13, 0.75);
  box-shadow: 0 5px #666;
  transform: translateY(4px);
}