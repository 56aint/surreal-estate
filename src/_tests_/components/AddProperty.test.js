import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../../components/AddProperty";

describe("AddProperty", () => {
  /* const mockProperty = {
    title: "1 bedroom house",
    type: "Detached",
    bathrooms: "3",
    bedrooms: "1",
    price: "450000",
    city: "Brighton",
    email: "no@no.email.com",
    userID: "userID",
    _id: "id",
    onSaveProperty: jest.fn(),
  }; */
  it("renders correctly", () => {
    const { asFragment } = render(<AddProperty />);
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });

  xit("renders the correct props", () => {
    const { getByText, getByTestId, getByRole } = render(<AddProperty />);
    expect(getByTestId("p_title-id")).toHaveClass("property-card_title");
    expect(getByTestId("p_type-id")).toHaveClass("property-card_type");
    expect(getByTestId("p_bathrooms-id")).toHaveClass(
      "property-card_bathrooms"
    );
    expect(getByTestId("p_bedrooms-id")).toHaveClass("property-card_bedrooms");
    expect(getByTestId("p_price-id")).toHaveClass("property-card_price");
    expect(getByTestId("p_email-id")).toHaveClass("property-card_email");
    expect(getByText("Save")).toHaveClass("save-button");
    expect(getByText("Save")).toBeTruthy();
    expect(getByRole("button")).toBeTruthy();
    expect(getByRole("button")).toHaveClass("save-button");
  });
});
