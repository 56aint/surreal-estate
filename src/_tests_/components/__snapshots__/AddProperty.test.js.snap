// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders correctly 1`] = `
<DocumentFragment>
  <div
    class="add-property"
    data-testid="Addproperty"
  >
    <form>
      <label
        class="form-label"
        data-testid="ap_title-id"
        for="title"
      >
        Title
        <input
          id="title"
          name="title"
          placeholder="Property Description"
          required=""
          type="text"
          value=""
        />
      </label>
      <label
        class="form-label"
        data-testid="ap_city-id"
        for="city"
      >
        <select
          id="city"
          name="city"
          required=""
        >
          <option
            value=""
          >
            Choose a city
          </option>
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Shefield"
          >
            Shefield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        class="form-label"
        data-testid="ap_type-id"
        for="type"
      >
        <select
          id="type"
          name="type"
          required=""
        >
          <option
            value=""
          >
            Choose a Property type
          </option>
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End of Terrace"
          >
            End of Terrace
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <label
        class="form-label"
        data-testid="ap_bedrooms-id"
        for="bedrooms"
      >
        Bedrooms
        <input
          class="number-input"
          id="bedrooms"
          name="bedrooms"
          required=""
          type="number"
          value=""
        />
      </label>
      <label
        class="form-label"
        data-testid="ap_bathrooms-id"
        for="bathrooms"
      >
        Bathrooms
        <input
          class="number-input"
          id="bathrooms"
          name="bathrooms"
          required=""
          type="number"
          value=""
        />
      </label>
      <label
        class="form-label"
        data-testid="ap_price-id"
        for="price"
      >
        Price
        <span
          class="gbp"
        >
          <input
            class="number-input"
            id="price"
            name="price"
            placeholder="Enter whole GBP (Â£)"
            required=""
            type="number"
            value=""
          />
        </span>
      </label>
      <br />
      <label
        class="form-label"
        data-testid="ap_email-id"
        for="email"
      >
        Email
        <input
          class="email-input"
          id="email"
          name="email"
          placeholder="your email"
          required=""
          type="email"
          value=""
        />
      </label>
      <button
        class="add-property-button"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
